metals/src/main/scala/scala/meta/metals/mtags/Mtags.scala

```diff
@@ -124,9 +124,11 @@ object Mtags extends LazyLogging {
   /** Index single Scala or Java from disk or zip file. */
   def index(fragment: Fragment): Document = {
     val uri = {
+      val base = fragment.base.toNIO.getFileName.toString
       // Need special handling because https://github.com/scalameta/scalameta/issues/1163
-      if (isZip(fragment.base.toNIO.getFileName.toString))
-        new URI(s"jar:${fragment.base.toURI.normalize()}!/${fragment.name}")
+      if (isZip(base) || isJar(base))
+        new URI(s"jar:${fragment.base.toURI
```

Comment:
Nitpick, when scalafmt produces weird output like this I tend to bind the splice into a variable `val baseFragment = fragment.base.toURI.normalize()` and the splice it with `$baseFragment`.